patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -2740,6 +2761,73 @@
  					}
  				}
  			}
+@@ -30113,7 +_,7 @@
+ 				Microsoft.Xna.Framework.Color color = inventoryBack;
+ 				Microsoft.Xna.Framework.Color color2 = new Microsoft.Xna.Framework.Color(80, 80, 80, 80);
+ 				int num39 = -1;
+-				for (int num40 = 0; num40 < 10; num40++) {
++				for (int num40 = 0; num40 < 3; num40++) {
+ 					if ((num40 == 8 && !flag4) || (num40 == 9 && !flag5))
+ 						continue;
+ 
+@@ -30134,6 +_,7 @@
+ 						context2 = 10;
+ 					}
+ 
++					/*
+ 					if (num39 == num38 && !_achievementAdvisor.CanDrawAboveCoins) {
+ 						_achievementAdvisor.DrawOneAchievement(spriteBatch, new Vector2(num41 - 10 - 47 - 47 - 14 - 14, num42 + 8), large: false);
+ 						UILinkPointNavigator.SetPosition(1570, new Vector2(num41 - 10 - 47 - 47 - 14 - 14, num42 + 8) + new Vector2(20f) * inventoryScale);
+@@ -30141,6 +_,7 @@
+ 
+ 					if (num39 == num37)
+ 						DrawDefenseCounter(num41, num42);
++					*/ 
+ 
+ 					Texture2D value4 = TextureAssets.InventoryTickOn.Value;
+ 					if (player[myPlayer].hideVisibleAccessory[num40])
+@@ -30187,10 +_,10 @@
+ 					player[myPlayer].mouseInterface = true;
+ 
+ 				num39 = -1;
+-				for (int num46 = 10; num46 < 20; num46++) {
++				for (int num46 = 10; num46 < 13; num46++) {
+ 					if ((num46 == 18 && !flag4) || (num46 == 19 && !flag5))
+ 						continue;
+-
++					
+ 					num39++;
+ 					bool num47 = LocalPlayer.IsAValidEquipmentSlotForIteration(num46);
+ 					flag3 = !num47;
+@@ -30228,7 +_,7 @@
+ 					player[myPlayer].mouseInterface = true;
+ 
+ 				num39 = -1;
+-				for (int num50 = 0; num50 < 10; num50++) {
++				for (int num50 = 0; num50 < 3; num50++) {
+ 					if ((num50 == 8 && !flag4) || (num50 == 9 && !flag5))
+ 						continue;
+ 
+@@ -30263,6 +_,19 @@
+ 				}
+ 
+ 				inventoryBack = color;
++				int chkMax = (int)((float)(num20) + (float)((5 + LocalPlayer.GetAmountOfExtraAccessorySlotsToShow() + 1) * 56) * Main.inventoryScale) + 4;
++				int randomLocationX = screenWidth - 64 - 28;
++
++				if (num39 == num38 && !_achievementAdvisor.CanDrawAboveCoins) {
++					_achievementAdvisor.DrawOneAchievement(spriteBatch, new Vector2(randomLocationX - 10 - 47 - 47 - 14 - 14, chkMax + 8), large: false);
++					UILinkPointNavigator.SetPosition(1570, new Vector2(randomLocationX - 10 - 47 - 47 - 14 - 14, chkMax + 8) + new Vector2(20f) * inventoryScale);
++				}
++
++				DrawDefenseCounter(randomLocationX, chkMax);
++
++				LoaderManager.Get<AccessorySlotLoader>().DrawAccSlots(num20);
```

Comment:
Personal request. It would be great if there was a way to get access to `num20` for my slot draw positions. Maybe store it in a global variable in `Main`?

Reply:
You can (and should) use the existing xLoc and yLoc properties to specify where you want the slot to be drawn - from there, AccessorySlot.DrawModded() will provide the exact positions it recommends through its position argument.

That said, after saying that and doublechecking the code, I can see how having that info might be helpful in deciding where to draw said slot less blindly.

Added AccessorySlotLoader.DrawVerticalAlignment and corresponding summary